frontend/src/pages/connectionTypes/manage/advanced/FileUploadAdvancedPropertiesForm.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import { Button, FormGroup } from '@patternfly/react-core';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+import { FileField } from '~/concepts/connectionTypes/types';
+import {
+  EXTENSION_REGEX,
+  isDuplicateExtension,
+} from '~/concepts/connectionTypes/fields/fieldUtils';
+import { AdvancedFieldProps } from '~/pages/connectionTypes/manage/advanced/types';
+import ExpandableFormSection from '~/components/ExpandableFormSection';
+import FileUploadExtensionRow from '~/pages/connectionTypes/manage/advanced/FileUploadExtensionRow';
+
+const FileUploadAdvancedPropertiesForm: React.FC<AdvancedFieldProps<FileField>> = ({
+  properties,
+  onChange,
+  onValidate,
+}) => {
+  const displayedExtensions = React.useMemo(
+    () => [...(properties.extensions?.length ? [...properties.extensions] : [''])],
+    [properties.extensions],
+  );
+  const lastTextRef = React.useRef<HTMLInputElement | null>(null);
+
+  React.useEffect(() => {
+    if (!properties.extensions?.length) {
+      onValidate(true);
+      return;
+    }
+    const valid = properties.extensions.every(
+      (extension, i) =>
+        !extension ||
+        (EXTENSION_REGEX.test(extension) && !isDuplicateExtension(i, properties.extensions || [])),
+    );
+    onValidate(valid);
+    // do not run when callback changes
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [properties.extensions]);
+
+  return (
+    <ExpandableFormSection
+      toggleText="Advanced settings"
+      initExpanded={!!properties.extensions}
```

Comment:
Do not init expanded if it's an empty array.
```suggestion
      initExpanded={!!properties.extensions?.length}
```